<launch>

<arg name = "map_size_x" default = "10.0"/>
<arg name = "map_size_y" default = "10.0"/>

<arg name = "start_x" default = "0.0"/>
<arg name = "start_y" default = "0.0"/>

<node pkg = "rviz" type = "rviz" name = "rviz" output = "screen" args="-d $(find grid_path_searcher)/launch/rviz_config/astar.rviz" required = "true"/>

    <node pkg = "grid_path_searcher" type = "astar_node" name = "astar_node" output = "screen" required = "true">
        <!--remap from = "~waypoints" to = "/waypoint_generator/waypoints"/-->
        <remap from = "~waypoints" to = "/goal"/>
        <remap from = "~map" to = "/random_complex/global_map"/>

        <param name = "map/margin" value = "0.0"/>
        <param name = "map/resolution" value = "0.2"/>
        <param name = "map/x_size" value = "$(arg map_size_x)"/>
        <param name = "map/y_size" value = "$(arg map_size_y)"/>

        <param name = "planning/start_x" value = "$(arg start_x)"/>
        <param name = "planning/start_y" value = "$(arg start_y)"/>
    </node>

    <node pkg ="grid_path_searcher" name ="random_complex" type ="random_complex" output = "screen">    
    
      <param name="init_state_x"   value="$(arg start_x)"/>
      <param name="init_state_y"   value="$(arg start_y)"/>

      <param name="map/x_size"     value="$(arg map_size_x)" />
      <param name="map/y_size"     value="$(arg map_size_y)" />

      <param name="map/circle_num" value="10"/>        
      <param name="map/obs_num"    value="70"/>        
      <param name="map/resolution" value="0.1"/>        

      <param name="ObstacleShape/lower_rad" value="0.1"/>
      <param name="ObstacleShape/upper_rad" value="0.7"/>

      <param name="CircleShape/lower_circle_rad"   value="0.6"/>        
      <param name="CircleShape/upper_circle_rad"   value="2.0"/>        
      
      <param name="sensing/rate"   value="0.5"/>        
    </node>

    <!--node pkg = "grid_path_searcher" name = "waypoint_generator" type = "waypoint_generator" output = "screen">
        <remap from = "~goal" to =  "/goal"/>
        
        <param name = "waypoint_type" value = "manual-lonely-waypoint"/>
    </node-->

</launch>